core/src/main/scala/chisel3/StrongEnum.scala

```diff
@@ -20,12 +20,12 @@ object EnumAnnotations {
     * @param target the enum instance being annotated
     * @param typeName the name of the enum's type (e.g. ''"mypackage.MyEnum"'')
     */
-  case class EnumComponentAnnotation(target: Named, enumTypeName: String) extends SingleTargetAnnotation[Named] {
+  case class EnumComponentAnnotation(target: Named, typeName: String) extends SingleTargetAnnotation[Named] {
     def duplicate(n: Named): EnumComponentAnnotation = this.copy(target = n)
   }
 
-  case class EnumComponentChiselAnnotation(target: InstanceId, enumTypeName: String) extends ChiselAnnotation {
-    def toFirrtl: EnumComponentAnnotation = EnumComponentAnnotation(target.toNamed, enumTypeName)
+  case class EnumComponentChiselAnnotation(target: InstanceId, typeName: String) extends ChiselAnnotation {
+    def toFirrtl: EnumComponentAnnotation = EnumComponentAnnotation(target.toNamed, typeName)
```

Comment:
I appreciate this change overall (especially the removal of snake_case 🙂 ), but I'm a bit less sure here. I agree this naming is better, but it's a source and binary incompatible change (the latter is less relevant since we break binary compatibility on major versions). There are a non-trivial number of things to do to ensure this is at least source compatible though and I'm just not sure it's worth it.

If you're interested I can let you know what to do, otherwise let's back this part of the change out and I'll merge the rest.

Reply:
I was careless about compatibility. 
I'll revert this change, for now.

Reply:
I have a question about scaladoc.
This change originally came from mismatching scaladoc's param name, so I want to change param name from typeName into enumTypeName.
However, in terms of compatibility, is this safe?

Reply:
> This change originally came from mismatching scaladoc's param name, so I want to change param name from typeName into enumTypeName.

Ah yes having those in sync is pretty important. There's no risk with changing the ScalaDoc, go ahead!